### PR TITLE
Full completion Add runInAction to be compliant with strict action mobx

### PR DESCRIPTION
**Pull request by jkeruzec**
#66

### CODE REVIEW

File: src/Base.ts [hunk 1]
```diff
@@ -7,8 +7,14 @@ import { action, observable, IObservableArray, runInAction } from 'mobx'
 
 export default class Base {
   @observable request: Request | null
-  @observable.shallow requests: IObservableArray<Request> = observable.array([])
+  @observable.shallow requests: IObservableArray<Request>
 
+  constructor() {
+      runInAction('init request', () => {
+          this.requests = observable.array([])
+        })
```
Comment (masylum) [inline src/Base.ts]: weird indentation

File: src/Base.ts [hunk 2]
```diff
@@ -29,7 +35,9 @@ export default class Base {
 
     const handledPromise = promise
       .then(response => {
-        if (this.request === request) this.request = null
+        if (this.request === request) runInAction('reset request', () => {
+            this.request = null
+          })
```
Comment (masylum) [inline src/Base.ts]: fix indentation:

```js
if (this.request === request) {
  runInAction('reset request', () => {
    this.request = null
  })
}
```

File: src/Base.ts [hunk 3]
```diff
@@ -48,10 +56,10 @@ export default class Base {
...
       abort
     })
-
-    this.request = request
-    this.requests.push(request)
-
+    runInAction('push request', () => {
+        this.request = request
+        this.requests.push(request)
+      })
```
Comment (masylum) [inline src/Base.ts]: indenation

File: src/Request.ts [hunk 4]
```diff
@@ -9,15 +9,17 @@ export default class Request {
...
+          this.progress = progress = 0
+          this.promise = promise
+      })
 
-    this.promise
-      .then(() => { this.state = 'fulfilled' })
-      .catch(() => { this.state = 'rejected' })
+      this.promise
+      .then(() => {  runInAction('change request state', () => this.state = 'fulfilled') })
+      .catch(() => { runInAction('change request state', () => this.state = 'rejected') })
```
Comment (masylum) [inline src/Request.ts]: we use two spaces for indentation :)

**Comment by mtvillwock** [comment]
@masylum @jkeruzec 
Is this still in line to be merged?

**Comment by masylum** [comment]
Being addressed on the latest version

**Pull request closed without merging**